assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceTilesWidget/AudienceTile/AudienceTilePagesMetric.js

```diff
@@ -125,34 +174,30 @@ export default function AudienceTilePagesMetric( {
 						/>
 					) }
 				</div>
-				<div className="googlesitekit-audience-segmentation-tile-metric__content">
-					{ ! hasDimensionValues && <AudienceTileNoData /> }
-					{ hasDimensionValues &&
-						validDimensionValues.map( ( content, index ) => {
-							return (
-								<div
-									key={ content?.value }
-									className="googlesitekit-audience-segmentation-tile-metric__page-metric-container"
-								>
-									<ContentLinkComponent content={ content } />
-									<div className="googlesitekit-audience-segmentation-tile-metric__page-metric-value">
-										{ numFmt(
-											topContent?.metricValues[ index ]
-												?.value
-										) }
-									</div>
-								</div>
-							);
-						} ) }
-					{ isMobileBreakpoint && isTopContentPartialData && (
-						<PartialDataNotice
-							content={ __(
-								'Still collecting full data for this timeframe, partial data is displayed for this metric',
-								'google-site-kit'
-							) }
-						/>
-					) }
-				</div>
+				<AudienceTilePagesMetricContent
+					topContentTitles={ topContentTitles }
+					topContent={ topContent }
+					isTopContentPartialData={ isTopContentPartialData }
+					hasCustomDimension={ ! hasMissingCustomDimension }
+					onCreateCustomDimension={ onCreateCustomDimension }
+					isSaving={ isSaving }
+				/>
+				{ customDimensionError && (
+					<AudienceErrorModal
```

Comment:
There's a bit of a glitch when showing the modal, as it appears while `isSaving` is true, so the Retry button initially shows up disabled with the spinner showing:

https://github.com/user-attachments/assets/a9f391b0-3169-42a7-9346-18788a8f2fe8

I've looked into it a bit - it's not the most straightforward to solve, with the logic split between here and `useCreateCustomDimensionForAudienceEffect()`, but I've got a potential solution which doesn't need much of a refactor. Here is it in diff form (with whitespace changes suppressed):

```diff
diff --git a/assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceTilesWidget/AudienceTile/AudienceTilePagesMetric.js b/assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceTilesWidget/AudienceTile/AudienceTilePagesMetric.js
index 8349a7c615..485cb7c907 100644
--- a/assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceTilesWidget/AudienceTile/AudienceTilePagesMetric.js
+++ b/assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceTilesWidget/AudienceTile/AudienceTilePagesMetric.js
@@ -100,6 +100,13 @@ export default function AudienceTilePagesMetric( {
 		)
 	);
 
+	const isRetryingCustomDimensionCreate = useSelect( ( select ) =>
+		select( CORE_FORMS ).getValue(
+			AUDIENCE_TILE_CUSTOM_DIMENSION_CREATE,
+			'isRetrying'
+		)
+	);
+
 	const propertyID = useSelect( ( select ) =>
 		select( MODULES_ANALYTICS_4 ).getPropertyID()
 	);
@@ -109,9 +116,11 @@ export default function AudienceTilePagesMetric( {
 	const { setPermissionScopeError, clearPermissionScopeError } =
 		useDispatch( CORE_USER );
 
-	const onCreateCustomDimension = useCallback( () => {
+	const onCreateCustomDimension = useCallback(
+		( { isRetrying } ) => {
 			setValues( AUDIENCE_TILE_CUSTOM_DIMENSION_CREATE, {
 				autoSubmit: true,
+				isRetrying,
 			} );
 
 			if ( ! hasAnalyticsEditScope ) {
@@ -130,12 +139,14 @@ export default function AudienceTilePagesMetric( {
 					},
 				} );
 			}
-	}, [
+		},
+		[
 			hasAnalyticsEditScope,
 			redirectURL,
 			setPermissionScopeError,
 			setValues,
-	] );
+		]
+	);
 
 	const onCancel = useCallback( () => {
 		setValues( AUDIENCE_TILE_CUSTOM_DIMENSION_CREATE, {
@@ -182,7 +193,8 @@ export default function AudienceTilePagesMetric( {
 					onCreateCustomDimension={ onCreateCustomDimension }
 					isSaving={ isSaving }
 				/>
-				{ customDimensionError && (
+				{ ( ( customDimensionError && ! isSaving ) ||
+					isRetryingCustomDimensionCreate ) && (
 					<AudienceErrorModal
 						apiErrors={ [ customDimensionError ] }
 						title={ __(
@@ -193,7 +205,9 @@ export default function AudienceTilePagesMetric( {
 							'Oops! Something went wrong. Retry enabling the metric.',
 							'google-site-kit'
 						) }
-						onRetry={ onCreateCustomDimension }
+						onRetry={ () =>
+							onCreateCustomDimension( { isRetrying: true } )
+						}
 						onCancel={ onCancel }
 						inProgress={ isSaving }
 					/>
diff --git a/assets/js/modules/analytics-4/hooks/useCreateCustomDimensionForAudienceEffect.js b/assets/js/modules/analytics-4/hooks/useCreateCustomDimensionForAudienceEffect.js
index 031e805b87..f821411810 100644
--- a/assets/js/modules/analytics-4/hooks/useCreateCustomDimensionForAudienceEffect.js
+++ b/assets/js/modules/analytics-4/hooks/useCreateCustomDimensionForAudienceEffect.js
@@ -80,6 +80,7 @@ export default function useCreateCustomDimensionForAudienceEffect() {
 
 			setValues( AUDIENCE_TILE_CUSTOM_DIMENSION_CREATE, {
 				isAutoCreatingCustomDimensionsForAudience: false,
+				isRetrying: false, // Reset retry state, if it was set.
 			} );
 		}
 		if ( isGA4Connected && hasAnalyticsEditScope && autoSubmit ) {

```

What do you think?

Reply:
Thanks so much, @techanvil. It works very well.